db.go

```diff
@@ -240,6 +245,94 @@ func newDBMetrics(db *DB, r prometheus.Registerer) *dbMetrics {
 	return m
 }
 
+// DBReadOnly provides APIs for read only operations on a database.
+type DBReadOnly struct {
+	db *DB
+}
+
+// OpenReadOnly returns a new DB in the given directory only for read operations.
+func OpenReadOnly(dir string, l log.Logger, r prometheus.Registerer) (db *DBReadOnly, err error) {
```

Comment:
This duplicates a fair bit of code compared to the main open function, how will we ensure they stay in sync?

Reply:
Don't think we need to as it has completely different purpose.